common/lib/dependabot/file_fetchers/base.rb

```diff
@@ -104,7 +108,16 @@ def target_branch
 
       sig { returns(T::Array[DependencyFile]) }
       def files
-        @files ||= T.let(fetch_files, T.nilable(T::Array[DependencyFile]))
+        return T.must(@files) if defined? @files
```

Comment:
Are both this and the `T.must` in the last line of the method needed?

Reply:
Yes but I think I can clean this up if I move the `@files` creation to the constructor.

Reply:
It doesn't seem to be the Ruby style (at least the Dependabot style?) to initialize all variables in the constructor but it makes using types easier.